Archive toplevel directory suffixed with tagname or commit id to match
the filename.

Created by:	Bernhard Froehlich <decke@bluelife.at>
--- routers/repo/repo.go.orig	2019-01-04 18:17:32 UTC
+++ routers/repo/repo.go
@@ -374,7 +374,7 @@ func Download(ctx *context.Context) {
 
 	archivePath = path.Join(archivePath, base.ShortSha(commit.ID.String())+ext)
 	if !com.IsFile(archivePath) {
-		if err := commit.CreateArchive(archivePath, archiveType); err != nil {
+		if err := commit.CreateArchive(archivePath, ctx.Repo.Repository.Name+"-"+refName, archiveType); err != nil {
 			ctx.ServerError("Download -> CreateArchive "+archivePath, err)
 			return
 		}
--- vendor/code.gitea.io/git/commit_archive.go.orig	2018-12-21 15:08:46 UTC
+++ vendor/code.gitea.io/git/commit_archive.go
@@ -6,8 +6,6 @@ package git
 
 import (
 	"fmt"
-	"path/filepath"
-	"strings"
 )
 
 // ArchiveType archive types
@@ -21,7 +19,7 @@ const (
 )
 
 // CreateArchive create archive content to the target path
-func (c *Commit) CreateArchive(target string, archiveType ArchiveType) error {
+func (c *Commit) CreateArchive(target string, prefix string, archiveType ArchiveType) error {
 	var format string
 	switch archiveType {
 	case ZIP:
@@ -32,6 +30,6 @@ func (c *Commit) CreateArchive(target string, archiveT
 		return fmt.Errorf("unknown format: %v", archiveType)
 	}
 
-	_, err := NewCommand("archive", "--prefix="+filepath.Base(strings.TrimSuffix(c.repo.Path, ".git"))+"/", "--format="+format, "-o", target, c.ID.String()).RunInDir(c.repo.Path)
+	_, err := NewCommand("archive", "--prefix="+prefix+"/", "--format="+format, "-o", target, c.ID.String()).RunInDir(c.repo.Path)
 	return err
 }
